2996_reflection/reflection.md

```diff
@@ -81,21 +110,73 @@ namespace std {
 
 In our initial proposal a value of type `std::meta::info` can represent:
 
-  - an error
+  - an error (corresponding to an "invalid reflection")
   - any (C++) type and type-alias
   - any function or member function
   - any namespace-scope variable or any C++ static data member
   - any non-static data member
   - any constant value
   - any template
+  - any namespace
 
 [ (DV) I'm currently excluding local variable representation.  We can change that if we have a use for them that we feel must be in the initial core functionality. ]
 
+
+The type `std::meta::info` is a _scalar_ type with associated namespace `std::meta`.  Nontype template arguments of type `std::meta::info` are permitted.
+The entity being reflected can affect the linkage of a template instance involving a reflection.  For example:
+
+```c++
+template<auto R> struct S {};
+
+extern int x;
+static int y;
+
+S<^x> sx;  // S<^x> has external name linkage.
+S<^y> sy;  // S<^y> has internal name linkage.
+```
+
+
 ## The Reflection Operator (`^`)
 
+The reflection operator produces a reflection value from a grammatical construct (its operand):
+
+> | _unary-expression_:
+> |       ...
+> |       `^` `::`
+> |       `^` _namespace-name_
+> |       `^` _type-id_
+> |       `^` _cast-expression_
+
+Note that _cast-expression_ includes _id-expression_, which in turn can designate templates, member names, etc.
+
+The current proposal requires that the _cast-expression_ be:
+
+  - a _primary-expression_ referring to a function or member function, or
+  - a _primary-expression_ referring to a namespace-scope variable or to a static data member, or
+  - a _primary-expression_ referring to a nonstatic data member, or
+  - a _primary-expression_ referring to a template, or
+  - a constant-expression.
+
+In a SFINAE context, a failure to substitute the operand of a reflection operator construct causes that construct to evaluate to an invalid reflection.
 
 ## Splicers (`[:`...`:]`)
 
+A reflection that is not an invalid reflection can be "spliced" into source code using one of several _splicer_ forms:
+
+ - `[: r :]` produces an _expression_ evaluating to the entity or constant value represented by `r`.
+ - `typename[: r :]` produces a _simple-type-specifier_ corresponding to the type represented by `r`.
```

Comment:
So even in a non-dependent content, I can't write like:
```cpp
using T = [: ^int :];
```
That's invalid, I _must_ write:
```cpp
using T = typename [: ^int :];
```
?

Reply:
We can probably allow the same implicit `typename` places as for dependent type names.